tests/integration/test_automation/testParser.py

```diff
@@ -0,0 +1,507 @@
+import yaml
+from pathlib import Path
+from colorama import Fore, Back, Style
+
+class ACE:
+
+    # constructor function    
+    def __init__(self, time_stamp,address,  snoop, rwtype, srcline):
+        self.time_stamp = time_stamp
+        self.address = address
+        self.snoop = snoop
+        self.rwtype = rwtype
+        self.srcline = srcline
+
+class AXI:
+
+    # constructor function    
+    def __init__(self, time_stamp,address, rwtype,srcline):
+        self.time_stamp = time_stamp
+        self.address = address
+        self.rwtype = rwtype
+        self.srcline = srcline
+
+class SNOOP:
+
+    # constructor function    
+    def __init__(self, time_stamp,address, crresp, msgtype, srcline):
+        self.time_stamp = time_stamp
+        self.address = address
+        self.crresp = crresp
+        self.msgtype = msgtype
+        self.srcline = srcline
+
+class DATA_BUFFER_ACCESS:
+
+    def __init__(self, address, min_address, max_address):
+        self.address = address
+        self.min_address = min_address
+        self.max_address = max_address
+
+class ACE_LIST_CHECKER:
+
+    def __init__(self, theList,theDict, rw):
+        self.theList = theList
+        self.theDict = theDict
+        self.rw = rw
+ 
+
+    def checker (self, group_index, message_index, certainty):
+        message_found = False
+        snoop_search = self.theDict[test_yaml['test']['groups'][i]['messages'][j][self.rw]]
+        for l in range(0,len(self.theList)):         
+            if self.theList[l].address == search_address and self.theList[l].snoop == snoop_search :
```

Comment:
where does search_address come from?

Reply:
It is a global that is defined before the object method is called. I can also pass in as an argument.

Reply:
done

Reply:
OK